### ISSUE TITLE
A posteriori chosen variances for alpha and beta (Kalman Linear Regression)

### ISSUE THREAD

**Issue by MMABK** [cantaro86/Financial-Models-Numerical-Methods#4]
Dear  cantaro86,

Could you, please, clarify how you chose  variances for alpha and beta. You chose the variances for  alpha and beta 1e-7 and 1e-2, correspondingly, by "looking at the plot". Do you estimate them in the following way:
(ret_test["ols_alpha"] - ret_test["ols_alpha"].shift(1)).iloc[1:].var()
and
(ret_test["ols_beta"] - ret_test["ols_beta"].shift(1)).iloc[1:].var()
?
In this case I got 0.0033 and 7e-8. Do you round those numbers to get 1e-7 and 1e-2?

**Comment by cantaro86** [comment]
Hi @MMABK 

Thank you for the question. What you asked is very useful, and I think I will add additional information in the notebook in order to clarify this choice. 

What I did is not rigorous and a wrong practice (looking ahead bias). 
My intention was just to provide an example of a good filter, and therefore I selected the variances such that the plots look good. This, of course, is cheating :)  
However, the heuristic argument (three-sigma rule) I used, is reasonable and correct, and actually there was no reason to adjust the parameters by looking at the plots. 

The method you suggest has the same problem (looking ahead bias). 
The test set cannot be used for parameter estimation. This is something I have to explain better in the notebook.
But you can use the training set!! 
The problem is that the error you obtain in this way depends on the time window (for a big time window you get a very small variance), and thus you have to be careful when you select the window.   

In the next days I will make some changes and then I will explain them here.

**Comment by MMABK** [comment]
Dear @cantaro86 
Thank you for the answer! I am looking forward to see the update.

Best,
Vladimir

**Comment by cantaro86** [comment]
Hey @MMABK 
I updated the notebook. There are no big changes, but I think now it is a bit clearer. 

As far as I know, there is no "standard" method to design a filter (estimate the parameters).  
I proposed a few methods, and one of them is the method you suggested above, where variances are computed for the rolling values. 
Introducing a validation set can be the best choice in this case. You can calculate the variance of the rolling values in the training set and run the filter in the validation set. If you obtain unreasonable values, you can repeat the estimation in the training set with a different rolling window.  

Please, let me know if now it is clear, and if you have any suggestions for improvements, or corrections. Otherwise we can close this issue.

Nicola

**Comment by MMABK** [comment]
Dear @cantaro86 

Thank you, it is clearer now.
